### PR TITLE
[MI-3286] Create an RHS state to connect user account and show different states for linked channels

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary

- Added a button for users to connect to MS Teams.
- Added different RHS states for showing linked channels.

image
image

### CODE REVIEW

File: webapp/src/types/module.d.ts
Comment (raghavaggarwal2308) [inline webapp/src/types/module.d.ts:1]: Just a question. What is the use of this file?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/api.go [hunk 1]
```diff
@@ -401,6 +406,12 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	a.p.API.PublishWebSocketEvent(
+		"connect",
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply server/api.go:412]: discussed

File: webapp/src/app.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    useApiRequestCompletionState({
+        serviceName: Constants.pluginApiServiceConfigs.needsConnect.apiServiceName,
+        handleSuccess: () => {
+            const data = needsConnectData as NeedsConnectData;
+            dispatch(setConnected(data.connected));
+        },
+    });
+
+    // This container is used just for making the API call to check user connection, it doesn't render anything.
+    return <></>;
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/app.tsx:30]: discussed

File: webapp/src/containers/Rhs/rhs.scss [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        &__title {
+            font-weight: 600;
+            size: 20px;
+            padding: 15px 0;
+        }
+    }
+}
+
+.msteams-sync-rhs-footer {
+    position: absolute;
```
[the remark this answers is not in the document]
Comment (ayusht2810) [reply webapp/src/containers/Rhs/rhs.scss]: it is removed in the next PR https://github.com/mattermost/mattermost-plugin-msteams-sync/blob/c41e673b6b1f3226296e435288863660a6ef8649/webapp/src/containers/Rhs/rhs.scss#L165

File: webapp/src/types/module.d.ts
Comment (ayusht2810) [reply webapp/src/types/module.d.ts:1]: this file is used to define the types present in the package

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by ayusht2810** [comment]
Closing this PR in favour of updated [figma designs](https://www.figma.com/file/aRgXgPZQ3lDaow5uatDzz9/MS-Teams-Sync-RHS?type=design&node-id=1-7&mode=design&t=ewtCZk3sFs0sMBLl-0). We will pick changes from this PR if required.

**Pull request closed without merging**